rclcpp/test/rclcpp/node_interfaces/test_node_parameters.cpp

```diff
@@ -28,31 +28,47 @@
 #include "rclcpp/node.hpp"
 #include "rclcpp/node_interfaces/node_parameters.hpp"
 
+#include "../../mocking_utils/patch.hpp"
+#include "../../utils/rclcpp_gtest_macros.hpp"
+
 class TestNodeParameters : public ::testing::Test
 {
 public:
   void SetUp()
   {
     rclcpp::init(0, nullptr);
+    rclcpp::NodeOptions options;
+    options.allow_undeclared_parameters(true);
+    node = std::make_shared<rclcpp::Node>("node", "ns", options);
+
+    // This dynamic cast is not necessary for the unittest itself, but instead is used to ensure
+    // the proper type is being tested and covered.
+    node_parameters =
+      dynamic_cast<rclcpp::node_interfaces::NodeParameters *>(
```

Comment:
@brawner I saw this while looking at a different pull request, can you explain why this is needed?

Are you trying to detect mocking or something? You know that `rclcpp::Node::get_node_parameter_interface()` will return a pointer to this class already. If the type of `node` was overridable I could understand maybe, but even then, I don't know why it should matter.

Reply:
It's not required, and if this line keeps getting copy/pasted into other places it should probably get removed. `get_node_parameters_interface()` returns a `NodeParametersInterface::SharedPtr`. Since the node_interface tests were testing the derived classes (`NodeParameters` in this case), I just decided to do a dynamic_cast to check the implemented class was the one I expected. The tests will work fine for NodeParametersInterface, since it's pure virtual anyway.

I did this in all the `node_interfaces` tests I added, so not just node_parameters.

Reply:
Ok, I was just curious, if this was in response to a specific issue.